### PR TITLE
Keep fixing the report integration

### PR DESCRIPTION
**Pull request by GaelVaroquaux**
Some more fixes to the report integration with sphinx-gallery.

CC @lucyleeow : we'll have to integrate this in sphinx-gallery in the near future, its just too difficult to get right. Let's discuss this soon.

### CODE REVIEW

**Comment by GaelVaroquaux** [comment]
With these fixes, the examples displays well.

**Comment by GaelVaroquaux** [comment]
Don't merge, I'm still on it: trying to get also the interactive views to display with sphinx-gallery.

**Comment by GaelVaroquaux** [comment]
This is now ready for merge: it captures all our interactive HTML objects and displays reasonably well.

This is super exciting! It will make our docs so much cooler. It will also help our design: I like thinking about UI based on examples.

File: nilearn/reporting/sphinx_report.py [hunk 1]
```diff
@@ -1,44 +1,48 @@
...
+            {0}
 
 '''  # noqa: E501
 
+def indent_and_espace(text, amount=12):
+    "Indent, skip empty lines, and escape string delimiters"
+    return (''.join(amount * ' ' + line.replace("'", '"')
+                for line in text.splitlines(True)
+                if line)
+            + 12 * ' ')
```
Comment (emdupre) [inline nilearn/reporting/sphinx_report.py]: Is there a reason this is not as above ?

```suggestion
            + amount * ' ')
```

**Review by emdupre — COMMENTED** [review]
Thanks so much ! :sparkles: 

Super cool to see this work with all the examples. I had one quick question before merging.

**Comment by GaelVaroquaux** [comment]
Good catch! This is why code review is awesome!

**Comment by emdupre** [comment]
Great, merging ! Thanks again !

**Pull request merged**